### PR TITLE
Add custom evaluator openvino clip models

### PR DESCRIPTION
**Pull request by pwolnows**

### CODE REVIEW

**Comment by openmodelzooci** [comment]
Can one of the admins verify this patch?

File: tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        Code source https://github.com/AlexKoff88/CLIP_benchmark/blob/openvino/clip_benchmark/models/openvino_clip.py
+        '''
+        features = self.image_encoder(image)
+        features = torch.from_numpy(features[self.image_encoder.output()])
+        return F.normalize(features, dim=-1) if normalize else features
+
+    def encode_text(self, text, normalize: bool = False):
+        # Code source https://github.com/AlexKoff88/CLIP_benchmark/blob/openvino/clip_benchmark/models/openvino_clip.py
+        x = self.torch_model.token_embedding(text)
+        x = x + self.torch_model.positional_embedding
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py:129]: is it possible replicate token embeddings into function avoiding loading whole pytorch model for evaluation as you use it only for data preprocessing

File: tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        x = x.permute(1, 0, 2)
+        x = torch.asarray(x).contiguous()
+        x = self.text_encoder((x, self.torch_model.attn_mask))
+        x = torch.from_numpy(x[self.text_encoder.output()])
+        x = x.permute(1, 0, 2)
+        x = self.torch_model.ln_final(x)
+        x = x[np.arange(x.shape[0]), text.argmax(dim=-1)] @ self.torch_model.text_projection
+        return F.normalize(x, dim=-1) if normalize else x
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py:137]: I do not think that it is good to introduce torch dependency for operations that can be done using numpy

File: tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import os
+import numpy as np
+import open_clip
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py:18]: please make packages that are not listed in default installation optional for import with try-except and UnsupportedPackages

Comment (pwolnows) [reply tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py:18]: will change

File: tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py (hunk 1 above)
Comment (pwolnows) [reply tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py]: Removed loading of pytorch model, instead embeddings are imported from npz file pointed in accuracy config file

File: tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py (hunk 2 above)
Comment (pwolnows) [reply tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/openvino_clip_evaluator.py]: Rewritten to not use torch

**Review by eaidova — APPROVED** [review]
